docs(installing): add instructions for using external Postgres db (#366)
btpworks/chronicle: docs/external-postgres.md

# Using an external database with the Chronicle Helm chart

Chronicle uses a PostgreSQL database to store data. By default, Chronicle will
use an embedded PostgreSQL database. However, you can configure Chronicle to
use an external PostgreSQL database instead.

## Configure Chronicle to use an external PostgreSQL database

The default values.yaml for the Chronicle helm chart configures an

> internal Postgres database.

Reviewer: ```suggestion
internal PostgreSQL database.
```
We switch back and forth somewhat, probably best to make this and the one below the same as the others.

Author: Thanks I've updated to match PostgreSQL everywhere

To use an external database, you must override these values in your
own values.yaml file.

> NOTE: The database must exist, Chronicle will not create it for you.

Reviewer: ```suggestion
NOTE: The database must exist, Chronicle will not create it for you.
Chronicle will create the necessary tables inside an empty database.
```

Author: Also added

The following example values.yaml file configures Chronicle to use
an external PostgreSQL database.

```yaml
postgres:
  enabled: false
  host: my-postgres-host
  port: 5432
  database: my-postgres-database
  user: my-postgres-username
  password: my-postgres-password
```

> Rather than setting the password in the values.yaml file, you can use a
> Kubernetes secret. For example:

Reviewer: ```suggestion
Rather than setting the password in the values.yaml file, it is
recommended that you use a Kubernetes secret. For example:
```

Author: Very good recomenataion, I've added it and reworded that section a little

internal PostgreSQL database.
To use an external database, you must override these values in your
own values.yaml file.

NOTE:   The database must exist, Chronicle will not create it for you.
        Chronicle will create the necessary tables inside an empty database.

The following example values.yaml file configures Chronicle to use
an external PostgreSQL database.

```yaml
postgres:
  enabled: false
  host: my-postgres-host
  port: 5432
  database: my-postgres-database
  user: my-postgres-username
  password: my-postgres-password
```

Rather than setting the password directly in the values.yaml file, it is
recommended that you use a Kubernetes secret containing the password.

The following example values.yaml file configures Chronicle to use an existing
secret containing the password for an external PostgreSQL database.

```yaml
postgres:
  enabled: false
  host: my-postgres-host
  port: 5432
  database: my-postgres-database
  user: my-postgres-username
  existingPasswordSecret: my-postgres-password-secret
  existingPasswordSecretKey: password
```

## Create a Kubernetes secret for the external database password

> Some Postgres helm charts will create a secret containing the database password.

Reviewer: ```suggestion
Some PostgreSQL helm charts will create a secret containing the database password.
```
(still to do this one, then we're good)

Author: Corrected now, sorry I could have sworn I'd done that one as well
